### DISCUSSION TITLE
not sure how to get channel id by youtube username

### DISCUSSION THREAD

**Discussion by rebelvg** [LuanRT/YouTube.js#823] [General]
hey! looking thru the API I cannot seem to find the option of getting "channel id" by youtube username.
for example, my channel url is https://www.youtube.com/@rebelvg and the youtube username is `rebelvg` or `@rebelvg`.
I want to get channel id of the user so I can make subsequent requests like "get channel" that uses channel id and so on.
how do I do that?
currently I'm leveraging google data api for that, but I want to completely ditch that one and use innertube for all operations.

**Comment by ChunkyProgrammer** [comment]
For this, you'd want to pass in the url to InnerTube's resolveURL function and then get the channel id from that. https://ytjs.dev/api/classes/Innertube.html#resolveurl

**Comment by rebelvg** [comment]
this seems to be working, thanks!
